walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,64 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
 
+/** Returns a collection of time ranges in which the meeting could fit */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> result = new ArrayList<TimeRange>();
+    boolean includeEvent;
+    boolean noEventsYet = true;
+    //Edgecase for too long of a meeting (takes more than a day)
+    if (request.getDuration() > TimeRange.getTimeInMinutes(23, 59)) {
+      return result;
+    }
+    //Initialization for lastEvent (used to compare with other events)
+    Event lastEvent = new Event("Not an event",
+    TimeRange.fromStartDuration(0, 30), Arrays.asList("Me"));
+    int lastTime;
+    for (Event event : events) {
+      includeEvent = false;
+      //Check if current event needs to be considered
+      for (String attendee : request.getAttendees()) {
+        if (event.getAttendees().contains(attendee)) {
+          includeEvent = true;
+          break;
+        }
+      }
+      if (includeEvent) {
+        if (noEventsYet) {
+          //Different case for the first event
```

Comment:
Can you elaborate a bit on what this means?

Reply:
I handled the first event differently because I could not use LastEvent, but when I reviewed this today I noticed its not necessary. I changed this and this conditional does not exist anymore